llvm/include/llvm/Analysis/IVDescriptors.h

```diff
@@ -236,10 +253,26 @@ class RecurrenceDescriptor {
     return Kind == RecurKind::IAnyOf || Kind == RecurKind::FAnyOf;
   }
 
+  /// Returns true if the recurrence kind is of the form
+  ///   select(cmp(),x,y) where one of (x,y) is increasing loop induction.
+  static bool isFindLastIVRecurrenceKind(RecurKind Kind) {
+    return Kind == RecurKind::IFindLastIV || Kind == RecurKind::FFindLastIV;
+  }
+
   /// Returns the type of the recurrence. This type can be narrower than the
   /// actual type of the Phi if the recurrence has been type-promoted.
   Type *getRecurrenceType() const { return RecurrenceType; }
 
+  /// Returns the sentinel value used to replace the start value.
+  Value *getSentinelValue() const {
+    if (isFindLastIVRecurrenceKind(Kind)) {
```

Comment:
Should this only ever be called for FindLastIV recurrences? Can it be na assert?

Reply:
Yes, it should be assertion.
9059eec7bc59ee4af1fa00ea0c66ac7145a5023e